Description: Link against mathicgb and friends as shared libraries again.
Author: Doug Torrance <dtorrance@debian.org>
Forwarded: TODO
Last-Update: 2021-07-13

--- a/M2/Macaulay2/e/Makefile.files
+++ b/M2/Macaulay2/e/Makefile.files
@@ -26,40 +26,6 @@
 	NCResolutions/nc-res-computation \
 	Polynomial \
 	monomial-collection \
-	mathicgb/mathicgb/CFile \
-	mathicgb/mathicgb/LogDomain \
-	mathicgb/mathicgb/LogDomainSet \
-	mathicgb/mathicgb/PolyRing \
-	mathicgb/mathicgb/Basis \
-	mathicgb/mathicgb/MonoLookup \
-	mathicgb/mathicgb/PolyBasis \
-	mathicgb/mathicgb/SparseMatrix \
-	mathicgb/mathicgb/QuadMatrix \
-	mathicgb/mathicgb/QuadMatrixBuilder \
-	mathicgb/mathicgb/Scanner \
-	mathicgb/mathicgb/ClassicGBAlg \
-	mathicgb/mathicgb/F4MatrixBuilder \
-	mathicgb/mathicgb/F4MatrixBuilder2 \
-	mathicgb/mathicgb/F4MatrixProjection \
-	mathicgb/mathicgb/F4MatrixReducer \
-	mathicgb/mathicgb/F4ProtoMatrix \
-	mathicgb/mathicgb/F4Reducer \
-	mathicgb/mathicgb/Reducer \
-	mathicgb/mathicgb/TypicalReducer \
-	mathicgb/mathicgb/ReducerDedup \
-	mathicgb/mathicgb/ReducerHash \
-	mathicgb/mathicgb/ReducerHashPack \
-	mathicgb/mathicgb/ReducerNoDedup \
-	mathicgb/mathicgb/ReducerPack \
-	mathicgb/mathicgb/ReducerPackDedup \
-	mathicgb/mathicgb/ModuleMonoSet \
-	mathicgb/mathicgb/PolyHashTable \
-	mathicgb/mathicgb/SPairs \
-	mathicgb/mathicgb/SigPolyBasis \
-	mathicgb/mathicgb/SigSPairQueue \
-	mathicgb/mathicgb/SigSPairs \
-	mathicgb/mathicgb/SignatureGB \
-	mathicgb/mathicgb/io-util \
 	eigen \
 	monomial-sets \
 	mutablecomplex \
@@ -212,18 +178,6 @@
 	weylalg
 
 COMMANDS = \
-	memtailor/memtailor \
-	memtailor/memtailor/Arena \
-	memtailor/memtailor/BufferPool \
-	memtailor/memtailor/MemoryBlocks \
-	mathic/mathic \
-	mathic/mathic/BitTriangle \
-	mathic/mathic/ColumnPrinter \
-	mathic/mathic/DivMask \
-	mathic/mathic/Timer \
-	mathic/mathic/display \
-	mathic/mathic/error \
-	mathicgb/mathicgb \
 	ntl-internal \
 	ntl-debugio \
 	matrix-kbasis \
@@ -252,26 +206,6 @@
 	franzi-interface
 
 NAMES_H = \
-	mathicgb/mathicgb/Atomic \
-	mathicgb/mathicgb/PrimeField \
-	mathicgb/mathicgb/MonoMonoid \
-	mathicgb/mathicgb/Range \
-	mathicgb/mathicgb/Poly \
-	mathicgb/mathicgb/FixedSizeMonomialMap \
-	mathicgb/mathicgb/KoszulQueue \
-	mathicgb/mathicgb/MathicIO \
-	mathicgb/mathicgb/MonoArena \
-	mathicgb/mathicgb/MonoOrder \
-	mathicgb/mathicgb/MonoProcessor \
-	mathicgb/mathicgb/MonomialMap \
-	mathicgb/mathicgb/NonCopyable \
-	mathicgb/mathicgb/RawVector \
-	mathicgb/mathicgb/ReducerHelper \
-	mathicgb/mathicgb/ScopeExit \
-	mathicgb/mathicgb/StaticMonoMap \
-	mathicgb/mathicgb/Unchar \
-	mathicgb/mathicgb/mtbb \
-	mathicgb/mathicgb/stdinc \
 	f4/f4-types \
 	f4/memblock \
 	VectorArithmetic \
--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1238,35 +1238,35 @@
      BUILTLIBS="-lreadline -lhistory $BUILTLIBS"
 fi
 
-dnl if test $BUILD_memtailor = no
-dnl then AC_LANG(C++)
-dnl      SAVE_CXXFLAGS="$CXXFLAGS"
-dnl      CXXFLAGS="$CXXFLAGS"
-dnl      AC_SEARCH_LIBS(MEMTAILOR_VERSION_STRING,memtailor,,BUILD_memtailor=yes)
-dnl      AC_CHECK_HEADER(memtailor.h,,BUILD_memtailor=yes)
-dnl      CXXFLAGS="$SAVE_CXXFLAGS"
-dnl fi
-dnl test $BUILD_memtailor = yes && BUILTLIBS="-lmemtailor $BUILTLIBS"
+if test $BUILD_memtailor = no
+then AC_LANG(C++)
+     SAVE_CXXFLAGS="$CXXFLAGS"
+     CXXFLAGS="$CXXFLAGS"
+     AC_SEARCH_LIBS(MEMTAILOR_VERSION_STRING,memtailor,,BUILD_memtailor=yes)
+     AC_CHECK_HEADER(memtailor.h,,BUILD_memtailor=yes)
+     CXXFLAGS="$SAVE_CXXFLAGS"
+fi
+test $BUILD_memtailor = yes && BUILTLIBS="-lmemtailor $BUILTLIBS"
 
-dnl if test $BUILD_mathic = no
-dnl then AC_LANG(C++)
-dnl      SAVE_CXXFLAGS="$CXXFLAGS"
-dnl      CXXFLAGS="$CXXFLAGS"
-dnl      AC_SEARCH_LIBS(MATHIC_VERSION_STRING,mathic,,BUILD_mathic=yes)
-dnl      AC_CHECK_HEADER(mathic.h,,BUILD_mathic=yes)
-dnl      CXXFLAGS="$SAVE_CXXFLAGS"
-dnl fi
-dnl test $BUILD_mathic = yes && BUILTLIBS="-lmathic $BUILTLIBS"
+if test $BUILD_mathic = no
+then AC_LANG(C++)
+     SAVE_CXXFLAGS="$CXXFLAGS"
+     CXXFLAGS="$CXXFLAGS"
+     AC_SEARCH_LIBS(MATHIC_VERSION_STRING,mathic,,BUILD_mathic=yes)
+     AC_CHECK_HEADER(mathic.h,,BUILD_mathic=yes)
+     CXXFLAGS="$SAVE_CXXFLAGS"
+fi
+test $BUILD_mathic = yes && BUILTLIBS="-lmathic $BUILTLIBS"
 
-dnl if test $BUILD_mathicgb = no
-dnl then AC_LANG(C++)
-dnl      SAVE_CXXFLAGS="$CXXFLAGS"
-dnl      CXXFLAGS="$CXXFLAGS"
-dnl      AC_CHECK_HEADER(mathicgb.h,,BUILD_mathicgb=yes; AC_MSG_NOTICE(will build mathicgb))
-dnl      AC_SEARCH_LIBS(MATHICGB_VERSION_STRING,mathicgb,,BUILD_mathicgb=yes; AC_MSG_NOTICE(will build mathicgb))
-dnl      CXXFLAGS="$SAVE_CXXFLAGS"
-dnl fi
-dnl test $BUILD_mathicgb = yes && BUILTLIBS="-lmathicgb $BUILTLIBS"
+if test $BUILD_mathicgb = no
+then AC_LANG(C++)
+     SAVE_CXXFLAGS="$CXXFLAGS"
+     CXXFLAGS="$CXXFLAGS"
+     AC_CHECK_HEADER(mathicgb.h,,BUILD_mathicgb=yes; AC_MSG_NOTICE(will build mathicgb))
+     AC_SEARCH_LIBS(MATHICGB_VERSION_STRING,mathicgb,,BUILD_mathicgb=yes; AC_MSG_NOTICE(will build mathicgb))
+     CXXFLAGS="$SAVE_CXXFLAGS"
+fi
+test $BUILD_mathicgb = yes && BUILTLIBS="-lmathicgb $BUILTLIBS"
 
 if test $BUILD_gtest = no
 then
--- a/M2/GNUmakefile.in
+++ b/M2/GNUmakefile.in
@@ -214,7 +214,7 @@
 ifeq "@DEVELOPMENT@" "yes"
 all-in-Macaulay2: all-in-submodules
 else
-all-in-Macaulay2: all-in-submodules git-checkout-in-memtailor git-checkout-in-mathic git-checkout-in-mathicgb
+all-in-Macaulay2: all-in-submodules
 endif
 
 ## 
